Source/MQTTnet/Server/MqttClientConnection.cs

```diff
@@ -159,13 +167,33 @@ private async Task<MqttClientDisconnectType> RunInternalAsync(MqttConnectionVali
                     Interlocked.Increment(ref _sentPacketsCount);
                     _lastPacketReceivedTimestamp = DateTime.UtcNow;
 
+                    if (packet is MqttAuthPacket clientAuthPacket)
+                    {
+                        if (_serverOptions.EnhancedAuthenticationBrokerHandler == null)
+                        {
+                            Session.WillMessage = null;
+                            disconnectType = MqttClientDisconnectType.Clean;
+                            await StopAsync();
```

Comment:
What do you think about this implementation? Is this the way to go, in order to cleanly disconnect?

Reply:
I am not sure if this is a "clean" situation. The problem is that the client is sending data which is not supported by the server. So in my mind this is a disconnect which is "not clean".

Reply:
>So in my mind this is a disconnect which is "not clean".

Makes absolutely sense, I'm going to change this to NotClean!